dashboard/modules/job/cli.py

```diff
@@ -217,6 +224,7 @@ def submit(
     entrypoint_memory: Optional[int],
     entrypoint_resources: Optional[str],
     no_wait: bool,
+    allow_exists: bool,
     verify: Union[bool, str],
     headers: Optional[str],
 ):
```

Comment:
This solution includes an additional request (job_exists), but I think the overhead is OK.

Reply:
> Maybe we can use get_job_info which will raise a RuntimeError if the job doesn't exist.

I initially started the change to use `get_job_info` to check if the job exists, but all the APIs today raise an exception when a job errors, as opposed to returning a valid response with status 404. If we use `get_job_info`, we're catching an exception anyways, so I decided to just catch the exception from `submit_job` instead.

Reply:
1. A 404 also makes sense to me, but `submit_job` is a public API, so I think we need to keep backward compatibility.

2. I prefer to avoid the case where the `submission_id` already exists, executing unnecessary code to make it more error-prone. That's why I suggest adding the check at the beginning of the function.

Reply:
I will try to figure out who is the best person to review this PR next Monday.

Reply:
thanks, I'll also look into just adding a new method `job_exists` to avoid changing any of the existing stable APIs